crates/typst/src/geom/color.rs

```diff
@@ -27,68 +43,429 @@ use crate::syntax::Spanned;
-    pub const SILVER: Self = Self::Rgba(RgbaColor::new(0xDD, 0xDD, 0xDD, 0xFF));
-    pub const WHITE: Self = Self::Rgba(RgbaColor::new(0xFF, 0xFF, 0xFF, 0xFF));
-    pub const NAVY: Self = Self::Rgba(RgbaColor::new(0x00, 0x1f, 0x3f, 0xFF));
-    pub const BLUE: Self = Self::Rgba(RgbaColor::new(0x00, 0x74, 0xD9, 0xFF));
-    pub const AQUA: Self = Self::Rgba(RgbaColor::new(0x7F, 0xDB, 0xFF, 0xFF));
-    pub const TEAL: Self = Self::Rgba(RgbaColor::new(0x39, 0xCC, 0xCC, 0xFF));
-    pub const EASTERN: Self = Self::Rgba(RgbaColor::new(0x23, 0x9D, 0xAD, 0xFF));
-    pub const PURPLE: Self = Self::Rgba(RgbaColor::new(0xB1, 0x0D, 0xC9, 0xFF));
-    pub const FUCHSIA: Self = Self::Rgba(RgbaColor::new(0xF0, 0x12, 0xBE, 0xFF));
-    pub const MAROON: Self = Self::Rgba(RgbaColor::new(0x85, 0x14, 0x4b, 0xFF));
-    pub const RED: Self = Self::Rgba(RgbaColor::new(0xFF, 0x41, 0x36, 0xFF));
-    pub const ORANGE: Self = Self::Rgba(RgbaColor::new(0xFF, 0x85, 0x1B, 0xFF));
-    pub const YELLOW: Self = Self::Rgba(RgbaColor::new(0xFF, 0xDC, 0x00, 0xFF));
-    pub const OLIVE: Self = Self::Rgba(RgbaColor::new(0x3D, 0x99, 0x70, 0xFF));
-    pub const GREEN: Self = Self::Rgba(RgbaColor::new(0x2E, 0xCC, 0x40, 0xFF));
-    pub const LIME: Self = Self::Rgba(RgbaColor::new(0x01, 0xFF, 0x70, 0xFF));
+    pub const BLACK: Self = Self::Luma(LumaColor(F32Scalar(0.0)));
+    pub const GRAY: Self = Self::Luma(LumaColor(F32Scalar(0.6666666)));
+    pub const WHITE: Self = Self::Luma(LumaColor(F32Scalar(1.0)));
+    pub const SILVER: Self = Self::Luma(LumaColor(F32Scalar(0.8666667)));
+    pub const NAVY: Self = Self::Rgba(RgbaColor::new(0.0, 0.121569, 0.247059, 1.0));
+    pub const BLUE: Self = Self::Rgba(RgbaColor::new(0.0, 0.454902, 0.85098, 1.0));
+    pub const AQUA: Self = Self::Rgba(RgbaColor::new(0.4980392, 0.858823, 1.0, 1.0));
+    pub const TEAL: Self = Self::Rgba(RgbaColor::new(0.223529, 0.8, 0.8, 1.0));
+    pub const EASTERN: Self =
+        Self::Rgba(RgbaColor::new(0.13725, 0.615686, 0.678431, 1.0));
+    pub const PURPLE: Self =
+        Self::Rgba(RgbaColor::new(0.694118, 0.050980, 0.788235, 1.0));
+    pub const FUCHSIA: Self =
+        Self::Rgba(RgbaColor::new(0.941177, 0.070588, 0.745098, 1.0));
+    pub const MAROON: Self =
+        Self::Rgba(RgbaColor::new(0.521569, 0.078431, 0.294118, 1.0));
+    pub const RED: Self = Self::Rgba(RgbaColor::new(1.0, 0.254902, 0.211765, 1.0));
+    pub const ORANGE: Self = Self::Rgba(RgbaColor::new(1.0, 0.521569, 0.105882, 1.0));
+    pub const YELLOW: Self = Self::Rgba(RgbaColor::new(1.0, 0.8627451, 0.0, 1.0));
+    pub const OLIVE: Self = Self::Rgba(RgbaColor::new(0.239216, 0.6, 0.4392157, 1.0));
+    pub const GREEN: Self = Self::Rgba(RgbaColor::new(0.1803922, 0.8, 0.2509804, 1.0));
+    pub const LIME: Self = Self::Rgba(RgbaColor::new(0.0039216, 1.0, 0.4392157, 1.0));
 
     /// Create a grayscale color.
     ///
+    /// A grayscale color is represented internally by an array of one components:
```

Comment:
```suggestion
    /// A grayscale color is represented internally by a single `lightness` component.
```
The array thing is weird here. I also think there is no need to link to the type's docs since the docs automatically do that from the `Component` type. Maybe `gray` in the signature and `lightness` here should agree on the name.